shared/teams/team/team-info.tsx

```diff
@@ -0,0 +1,157 @@
+  const teamMeta = Container.useSelector(s => Constants.getTeamMeta(s, teamID))
+  const teamDetails = Container.useSelector(s => Constants.getTeamDetails(s, teamID))
+
+  const teamname = teamMeta.teamname
+  const lastDot = teamname.lastIndexOf('.')
+  const isSubteam = lastDot !== -1
+  const _leafName = isSubteam ? teamname.substr(lastDot + 1) : teamname
+  const parentTeamNameWithDot = isSubteam ? teamname.substr(0, lastDot + 1) : undefined
+
+  const [newName, _setName] = React.useState(_leafName)
+  const setName = (newName: string) => _setName(newName.replace(/[^a-zA-Z0-9_]/, ''))
+  const [description, setDescription] = React.useState(teamDetails.description)
+
+  const saveDisabled =
+    (description === teamDetails.description && newName === _leafName) || newName.length < 3
+  const waiting = Container.useAnyWaiting(Constants.teamWaitingKey(teamID), Constants.teamRenameWaitingKey)
+
+  const errors = [
+    ['rename', Container.useAnyErrors(Constants.teamRenameWaitingKey)?.message],
+    ['desc', Container.useSelector(state => state.teams.errorInEditDescription)],
+  ]
+
+  const onBack = () => dispatch(nav.safeNavigateUpPayload())
+  const onSave = () => {
+    if (newName !== _leafName) {
+      dispatch(TeamsGen.createRenameTeam({newName: parentTeamNameWithDot + newName, oldName: teamname}))
+    }
+    if (description !== teamDetails.description) {
+      dispatch(TeamsGen.createEditTeamDescription({description, teamID}))
+    }
+  }
+  const onEditAvatar = () =>
+    dispatch(
+      nav.safeNavigateAppendPayload({
+        path: [{props: {sendChatNotification: true, showBack: true, teamID}, selected: 'profileEditAvatar'}],
+      })
+    )
+  return (
+    <Kb.Modal
+      mode="DefaultFullHeight"
+      onClose={onBack}
+      header={{
+        leftButton: <Kb.Icon type="iconfont-arrow-left" onClick={onBack} />,
+        title: <ModalTitle teamID={teamID} title={isSubteam ? 'Edit subteam info' : 'Edit team info'} />,
+      }}
+      footer={{
+        content: (
+          <Kb.Button
+            label="Save"
+            onClick={onSave}
+            fullWidth={true}
+            disabled={saveDisabled}
+            waiting={waiting}
+          />
+        ),
+      }}
+      banners={errors
+        .filter(e => !!e[1])
+        .map(e => (
+          <Kb.Banner color="red" key={e[0]}>
+            {e[1]!}
+          </Kb.Banner>
+        ))}
+      allowOverflow={true}
+      backgroundStyle={styles.bg}
+    >
+      <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true} style={styles.body} gap="tiny">
+        <Kb.Avatar
+          editable={true}
+          onEditAvatarClick={onEditAvatar}
+          teamname={teamname}
+          size={96}
+          style={styles.avatar}
+        />
+        {isSubteam ? (
+          <Kb.NewInput
+            autoFocus={true}
+            maxLength={16}
+            onChangeText={setName}
+            prefix={parentTeamNameWithDot}
```

Comment:
can we make this look disabled-like? 
<img width="380" alt="image" src="https://user-images.githubusercontent.com/11968340/80530008-6311c580-8966-11ea-8b35-56b97aa388aa.png">

Reply:
I've had a rough time with this thing, also couldn't get a label to show. I feel like we should combine `LabeledInput` and `NewInput`, otherwise we have to continually add complexity to both of them.